### PR TITLE
QGA-9 Create quizzes routes

### PR DESCRIPTION
**Pull request by sadinovsr**

### CODE REVIEW

**Comment by Gruseckis** [comment]
You will need to resolve merge conflicts before merging

File: controllers/quizController.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
+import * as QuizModel from '../models/QuizModel';
+import AppError from '../errors/AppError';
+
+const getQuizzes = async (req, res, next) => {
+  try {
+    const quizzes = await QuizModel.getAllQuizzes();
+    res.status(200).send({
+      payload: { quizzes }
```
Comment (Gruseckis) [inline controllers/quizController.js:8]: please return empty array if no quizzes were found

File: controllers/quizController.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    next(new AppError(error.message));
+  }
+}
+
+const addQuiz = async (req, res, next) => {
+  try {
+    const quiz = await QuizModel.save({
+      ownerId: req.body.userId,
+      name: req.body.name,
+      description: req.body.description
```
Comment (Gruseckis) [inline controllers/quizController.js:20]: Questions are missing, but we need to check how quiz saving will be handled by front end

File: controllers/quizController.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  }
+}
+
+const deleteQuiz = async (req, res, next) => {
+  try {
+    const quiz = await QuizModel.getQuizById(req.params.quizId);
+    if (!quiz) {
+      throw new AppError('Cannot delete quiz which does not exist');
+    }
+    await QuizModel.deleteQuizById(req.params.quizId);
```
Comment (Gruseckis) [inline controllers/quizController.js]: delete quiz by id should return deleted quiz so previous check is redundant.

File: server.js [hunk 4]
```diff
@@ -5,6 +5,7 @@ import defaultErrorHandler from './middlewares/defaultErrorHandler';
 import mongoose from 'mongoose';
 import bodyParser from 'body-parser';
 import index from './routes/index';
+import quiz from './routes/quiz';
```
Comment (Gruseckis) [inline server.js]: Please give it more meaningful name like quizRoutes

**Review by Gruseckis — CHANGES_REQUESTED** [review]

File: controllers/quizController.js (hunk 1 above)
Comment (sadinovsr) [reply controllers/quizController.js:8]: model.find() already returns empty array by default if nothing found.

Comment (Gruseckis) [reply controllers/quizController.js:8]: Ok. Did not know that

**Review by Gruseckis — APPROVED** [review]

**Pull request merged**